scanpy/preprocessing/_simple.py

```diff
@@ -294,8 +300,12 @@ def log1p(
     def _log1p(X):
         if issparse(X):
             np.log1p(X.data, out=X.data)
+            if base is not None:
+                np.divide(X.data, np.log(base), out=X.data)
         else:
             np.log1p(X, out=X)
+            if base is not None:
+                np.divide(X, np.log(base), out=X)
         return X
```

Comment:
This is most likely 2×–3× as fast when using log2(X)÷log2(base). We should

1. investigate if that’s the case
2. still use log1p if base is None because it’s probably more accurate

Reply:
As a heads up, there's a good chance this will differ depending on how numpy was built.

Reply:
Which ways of building it do you mean? Blas or CPU extensions or what?

log2 is probably always faster because you simply chop off the exponent part of the float, subtract 1 and return it.

Reply:
Both I guess. MKL vs. OpenBLAS vs. whatever your HPC does can make a large difference. Ability to vectorize seems like a large factor.

This function is also not a bottleneck for typical usage, so I'm not too worried about performance. If this were an implementation for numpy, then it'd be more of an issue.

Reply:
SIMD and AVX2 are in basically everything today, but you’re right. Let’s not do premature optimization.